addons/remix-tree-button/main.js

```diff
@@ -1,35 +1,121 @@
+
+  function goToRemixTree(e) {
+    if (addon.settings.get("new-tab") || e.ctrlKey || e.metaKey) {
+      window.open(appendToProjectURL("remixtree"), "_blank", "noopener,noreferrer");
+    } else {
+      window.location.href = appendToProjectURL("remixtree");
+    }
+  }
+
+  function setOpensInNewTab(newTab, element) {
+    if (newTab) {
+      element.target = "_blank";
+      element.rel = "noopener noreferrer";
+    } else {
+      element.removeAttribute("target");
+      element.removeAttribute("rel");
+    }
+  }
+
+  // -- Handle show/hide --
+
+  async function setButton(enabled) {
+    // Add a button next to "Copy Link"
+    if (enabled) {
+      if (document.querySelector("#scratchAddonsRemixTreeBtn")) {
+        // If it's already there...
+        document.querySelector("#scratchAddonsRemixTreeBtn").style.display = "block";
+      } else {
+        addon.tab
+          .waitForElement(".flex-row.subactions", {
+            reduxCondition: (state) => state.scratchGui.mode.isPlayerOnly,
+          })
+          .then(() => {
+            if (!document.querySelector(".copy-link-button")) return;
+            if (document.querySelector("#scratchAddonsRemixTreeBtn")) return; // Check again because we're inside a promise
+            const remixtree = document.createElement("button");
+
+            const remixtreeSpan = document.createElement("span");
+            remixtreeSpan.innerText = msg("remix-tree");
+            remixtree.className = "button action-button remixtree-button";
+            remixtree.id = "scratchAddonsRemixTreeBtn";
+            remixtree.appendChild(remixtreeSpan);
+            remixtree.addEventListener("click", (e) => {
+              goToRemixTree(e);
+            });
+            addon.tab.appendToSharedSpace({ space: "afterCopyLinkButton", element: remixtree, order: 0 });
           });
-          addon.tab.appendToSharedSpace({ space: "afterCopyLinkButton", element: remixtree, order: 0 });
-        });
+      }
+    } else {
+      if (document.querySelector("#scratchAddonsRemixTreeBtn")) {
+        document.querySelector("#scratchAddonsRemixTreeBtn").style.display = "none";
+      }
     }
   }
 
-  loadRemixButton();
-  addon.tab.addEventListener("urlChange", () => {
-    loadRemixButton();
-  });
+  async function setViewAllLink(enabled) {
+    // Change the "View all" link to .../remixes to link to .../remixtree
+    addon.tab
+      .waitForElement(".remix-list", {
+        reduxCondition: (state) => state.scratchGui.mode.isPlayerOnly,
+      })
+      .then(() => {
+        if (enabled) {
+          const link = document.querySelector(".remix-list a");
+          const linkSpan = link.firstChild;
+
+          if (!originalTextContent) {
+            originalTextContent = linkSpan.textContent; // Store what the button's text content was before the first time we change it (because different languages).
```

Comment:
If you want, you can remove the global variable and just use `addon.tab.scratchMessage("project.viewAllInList")`